reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -331,46 +524,74 @@ wrap e cfg = do
-              ref <- newIORef $ Just t
-              writeChan events [EventTriggerRef ref :=> TriggerInvocation v (return ())])
-  return $ Element
-    { _element_events = es
-    , _element_raw = e
-    }
-
-{-# INLINABLE makeElement #-}
-makeElement :: forall er t m a. (MonadJSM m, MonadFix m, MonadReflexCreateTrigger t m, Adjustable t m) => Text -> ElementConfig er t GhcjsDomSpace -> ImmediateDomBuilderT t m a -> ImmediateDomBuilderT t m ((Element er GhcjsDomSpace t, a), DOM.Element)
-makeElement elementTag cfg child = do
-  doc <- askDocument
-  e <- liftJSM $ uncheckedCastTo DOM.Element <$> case cfg ^. namespace of
-    Nothing -> createElement doc elementTag
-    Just ens -> createElementNS doc (Just ens) elementTag
-  ImmediateDomBuilderT $ iforM_ (cfg ^. initialAttributes) $ \(AttributeName mAttrNamespace n) v -> case mAttrNamespace of
-    Nothing -> lift $ setAttribute e n v
-    Just ans -> lift $ setAttributeNS e (Just ans) n v
-  result <- flip localEnv child $ \env -> env
-    { _immediateDomBuilderEnv_parent = toNode e
-    }
-  append $ toNode e
-  wrapped <- wrap e $ extractRawElementConfig cfg
-  return ((wrapped, result), e)
+  return eventTriggerRefs
+
+{-# SPECIALIZE wrap
+  :: Chan [DSum (EventTriggerRef Spider) TriggerInvocation]
+  -> DOM.Element
+  -> RawElementConfig er Spider HydrationDomSpace
+  -> HydrationDomBuilderT HydrationDomSpace Spider HydrationM (DMap EventName (EventFilterTriggerRef Spider er))
+  #-}
+
+{-# SPECIALIZE wrap
+  :: Chan [DSum (EventTriggerRef Spider) TriggerInvocation]
+  -> DOM.Element
+  -> RawElementConfig er Spider GhcjsDomSpace
+  -> HydrationDomBuilderT GhcjsDomSpace Spider HydrationM (DMap EventName (EventFilterTriggerRef Spider er))
+  #-}
+
+{-# INLINE triggerBody #-}
+triggerBody
+  :: forall s er t x. EventSpec s ~ GhcjsEventSpec
+  => DOM.JSContextRef
+  -> RawElementConfig er t s
+  -> Chan [DSum (EventTriggerRef t) TriggerInvocation]
+  -> DMap EventName (EventFilterTriggerRef t er)
+  -> DOM.Element
+  -> WrapArg er EventName x
+  -> EventTrigger t x
+  -> IO (IO ())
+triggerBody ctx cfg events eventTriggerRefs e (WrapArg en) t = case DMap.lookup en eventTriggerRefs of
+  Just (EventFilterTriggerRef r) -> do
+    writeIORef r $ Just t
+    return $ do
+      writeIORef r Nothing
+  Nothing -> (`runJSM` ctx) <$> (`runJSM` ctx) (elementOnEventName en e $ do
+    evt <- DOM.event
+    mv <- lift $ unGhcjsEventHandler handler (en, GhcjsDomEvent evt)
+    case mv of
+      Nothing -> return ()
+      Just v -> liftIO $ do
+        --TODO: I don't think this is quite right: if a new trigger is created between when this is enqueued and when it fires, this may not work quite right
+        ref <- newIORef $ Just t
+        writeChan events [EventTriggerRef ref :=> TriggerInvocation v (return ())])
+  where
+    -- Note: this needs to be done strictly and outside of the newFanEventWithTrigger, so that the newFanEventWithTrigger doesn't
```

Comment:
TODO: The requirements of this note are not met, and I believe this is still true